lib/services/azuresqldb/client.js

```diff
@@ -0,0 +1,244 @@
+/* jshint camelcase: false */
+/* jshint newcap: false */
+var HttpStatus = require('http-status-codes');
+var request = require('request');
+var async = require('async');
+var util = require('util');
+
+var API_VERSION_FOR_TOKEN = '2015-05-01-preview';
+var API_VERISON_FOR_RESOURCE_GROUP = '2015-11-01';
+var API_VERSION_FOR_SQL = '2014-04-01-preview';
+var AZURE_RESOURCE_MANAGER_ENDPOINT = 'https://management.azure.com/';
+var MOONCAKE_RESOURCE_MANAGER_ENDPOINT = 'https://management.chinacloudapi.cn/';
+var AZURE_AD_ENDPOINT = 'https://login.windows.net';
+var MOONCAKE_AD_ENDPOINT = 'https://login.chinacloudapi.cn';
+
+var sqldbOperations = function (log, azure) {
+    this.log = log;
+    this.azure = azure;
+    this.resourceManagerEndpointUrl = (azure.environment == 'AzureChinaCloud') ? MOONCAKE_RESOURCE_MANAGER_ENDPOINT : AZURE_RESOURCE_MANAGER_ENDPOINT;
+    this.activeDirectoryEndpointUrl = (azure.environment == 'AzureChinaCloud') ? MOONCAKE_AD_ENDPOINT : AZURE_AD_ENDPOINT;
+
+    log.info('sqldb client CTOR');
+
+    this.POST = function (url, headers, data, apiVersion, useForm, callback) {
+        request({
+            url: url,
+            qs: { 'api-version': apiVersion },
+            method: 'POST',
+            headers: headers,
+            form: useForm ? data : null,
+            json: useForm ? null : data
+        }, function (err, response, body) {
+            //if (err) retry?
+            callback(err, response, body);
+        });
+    };
+
+    this.PUT = function (url, headers, data, apiVersion, callback) {
+        request({
+            url: url,
+            qs: { 'api-version': apiVersion },
+            method: 'PUT',
+            headers: headers,
+            json: data
+        }, function (err, response, body) {
+            //if (err) retry?
```

Comment:
I think backlog.  There are a number of things I want to do, but I don't want to push this PR too far.  I'll continue work on backlog as soon as this PR is done.